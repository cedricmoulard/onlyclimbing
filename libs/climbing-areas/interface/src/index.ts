export * from './lib/route-interface';
export * from './lib/route-level-enum';
export * from './lib/climbing-area-interface';
