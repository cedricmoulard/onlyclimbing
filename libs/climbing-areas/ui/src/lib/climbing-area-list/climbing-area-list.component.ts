import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
import { ClimbingArea } from '@oc/climbing-areas/interface';

@Component({
  selector: 'oc-climbing-area-list',
  template: `
    <ul>
      <li
        *ngFor="let climbingArea of climbingAreas"
        (click)="selected.emit(climbingArea)"
        [id]="climbingArea.id"
      >
        {{ climbingArea.name }}
      </li>
    </ul>
  `,
  styles: []
})
export class ClimbingAreaListComponent {
  @Input() climbingAreas: ClimbingArea[];
  @Output() selected: EventEmitter<ClimbingArea>;

  constructor() {
    this.selected = new EventEmitter<ClimbingArea>();
  }
}
