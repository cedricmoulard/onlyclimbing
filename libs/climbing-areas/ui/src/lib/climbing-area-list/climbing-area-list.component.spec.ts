import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { ClimbingAreaListComponent } from './climbing-area-list.component';
import { By } from '@angular/platform-browser';
import { ClimbingArea } from '@oc/climbing-areas/interface';

describe('ClimbingAreaListComponent', () => {
  let component: ClimbingAreaListComponent;
  let fixture: ComponentFixture<ClimbingAreaListComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ClimbingAreaListComponent]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ClimbingAreaListComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('should display climbing areas', () => {
    component.climbingAreas = [
      { id: '1', name: 'Area 1' },
      { id: '2', name: 'Area 2' },
      { id: '3', name: 'Area 3' }
    ];
    fixture.detectChanges();

    const elements = fixture.debugElement.queryAll(By.css('li'));

    expect(elements.length).toBe(3);
  });

  test('should display climbing areas name', () => {
    component.climbingAreas = [{ id: '1', name: 'Area 1' }];
    fixture.detectChanges();

    const element = fixture.debugElement.query(By.css('li'));
    expect(element.nativeElement).toMatchSnapshot();
  });

  test('should emit event on click', () => {
    const climbingArea: ClimbingArea = { id: '1', name: 'Area 1' };

    const emitMock = spyOn(component.selected, 'emit');

    component.climbingAreas = [climbingArea];
    fixture.detectChanges();

    const element = fixture.debugElement.query(By.css('li'));
    element.triggerEventHandler('click', null);

    expect(emitMock).toBeCalledTimes(1);
    expect(emitMock).toBeCalledWith(climbingArea);
  });
});
