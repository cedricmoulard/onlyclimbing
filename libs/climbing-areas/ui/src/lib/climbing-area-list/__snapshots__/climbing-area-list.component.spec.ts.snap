// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClimbingAreaListComponent should display climbing areas name 1`] = `
<li
  id="1"
>
  Area 1
</li>
`;
